### PR TITLE
[Storage] Add 'create_if_not_exists()' for BlobContainerClient, FileShareClient, QueueClient, FileSystemClient

### PR DESCRIPTION
**Pull request by vincenttran-msft**
As raised by a customer in #21044 and eventually added as a feature request, other SDKs have the functionality `create_if_not_exists()` but is missing in the Python-SDK. 

This PR adds such functionality for the `BlobContainerClient`, `FileShareClient`, `QueueClient`, and `FileSystemClient` (datalake).

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,11 @@
 ## 12.6.1 (Unreleased)
 
 ### Features Added
-
+- Added support for `create_if_not_exists()` for the following clients:
+  - `BlobContainerClient`
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/CHANGELOG.md:7]: You should update each package's `CHANGELOG` individually with only the changes relevant to that that package. For example, this is the `CHAGENLOG` for datalake so it should just have ``Added support for `create_if_not_exists()` for `FileSystemClient`.`` Then you'd update each other package's `CHANGELOG` with the relevant client.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py [hunk 2]
```diff
@@ -264,6 +265,28 @@ def create_file_system(self, metadata=None,  # type: Optional[Dict[str, str]]
...
+
+        :keyword Dict(str,str) metadata:
+            A dict with name-value pairs to associate with the
+            file system as metadata. Example: `{'Category':'test'}`
+        :keyword public_access:
+            To specify whether data in the file system may be accessed publicly and the level of access.
+        :type public_access: ~azure.storage.filedatalake.PublicAccess
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py:282]: I think we talked about this but `create_file_system` says it returns a `FileSystemClient`. Is that actually true? If so, this return type of should `~azure.storage.filedatalake.FileSystemClient or None`

File: sdk/storage/azure-storage-blob/tests/recordings/test_container.test_create_container_with_public_access_container.yaml [hunk 3]
```diff
@@ -3,21 +3,21 @@ interactions:
     body: null
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/tests/recordings/test_container.test_create_container_with_public_access_container.yaml:3]: Any reason you re-recorded this test? Probably can revert this?

Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/tests/recordings/test_container.test_create_container_with_public_access_container.yaml:3]: This comment applies to a couple of other tests in here too. If you did not change the test or the code the test is testing, no need to check in a new recording.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py [hunk 4]
```diff
@@ -371,6 +371,39 @@ def create_share(self, **kwargs):
...
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :keyword protocols:
+            Protocols to enable on the share. Only one protocol can be enabled on the share.
+        :keyword protocols: str or ~azure.storage.fileshare.ShareProtocols
+        :keyword root_squash:
+            Root squash to set on the share.
+            Only valid for NFS shares. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'.
+        :keyword root_squash: str or ~azure.storage.fileshare.ShareRootSquash
+        :rtype: None
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py:400]: Similar question about return type here. `create_share` claims to return a `dict`.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py (hunk 2 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py:282]: I believe the answer is no, and that `create_file_system` wrongly claims to return a `FileSystemClient`. `create_file_system` returns a call into `self._container_client.create_container`, which then returns a call to `self._client.container.create`, which finally does not call into any other methods and has a return of `None, or the result of cls(response)` (verified by skimming through the code & not just looking at the docstring)

So with this finding, should I also update the docstring of `create_file_system` to correctly represent its return-type? Thanks!

File: sdk/storage/azure-storage-blob/tests/recordings/test_container.test_create_container_with_public_access_container.yaml (hunk 3 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/tests/recordings/test_container.test_create_container_with_public_access_container.yaml:3]: I likely ran other tests outside of the one's I introduced and accidently added new recordings, will revert!

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py (hunk 4 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py:400]: I believe same findings as previously (that `create_share` falsely claims to return a `dict`).
`create_share` returns a call to `self._client.share.create`, whose return is `None, or the result of cls(response)`.

Same as previously commented, should I include the docstring changes in this PR?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py (hunk 2 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py:282]: Thanks for fixing this!

File: sdk/storage/azure-storage-file-share/tests/recordings/test_share_async.test_create_share_async.yaml [hunk 5]
```diff
@@ -2,102 +2,28 @@ interactions:
 - request:
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/tests/recordings/test_share_async.test_create_share_async.yaml:2]: Looks like one pesky recording missed in the revert :)

**Review by jalauzon-msft — COMMENTED** [review]
Looks good! Just one last recording to revert and still some failing tests in ci

File: sdk/storage/azure-storage-file-share/tests/recordings/test_share_async.test_create_share_async.yaml (hunk 5 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-share/tests/recordings/test_share_async.test_create_share_async.yaml:2]: So close.... thanks for pointing it out! :)

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**